common/jest-eslint.config.mjs

```diff
@@ -0,0 +1,32 @@
+import tsEslintConfig from './ts-eslint.config.mjs';
+import { createRequire } from 'module';
+const require = createRequire(import.meta.url);
+
+const tsEslint = require('typescript-eslint');
+const jest = require('eslint-plugin-jest');
+
+const jestEslintConfig = tsEslint.config(
+  tsEslintConfig,
+  {
+    plugins: jest.configs['flat/recommended']['plugins'],
+  },
+  {
+    languageOptions: {
+      globals: jest.configs['flat/recommended']['languageOptions']['globals'],
+    },
+  },
+  {
+    rules: {
+      ...jest.configs['flat/recommended']['rules'],
+    },
+  },
+  {
+    settings: {
+      jest: {
+        version: '29.7.0',
```

Comment:
I tried to determine this version dynamically, but it doesn't work in the `pre-commit.ci` run. There's probably some automation we can do with the dependency updates script to notice a change and either directly update here or at least leave a PR comment letting us know that it needs to be updated.